core/src/main/java/com/sequenceiq/cloudbreak/service/TlsSecurityService.java

```diff
@@ -35,23 +40,39 @@
     @Inject
     private InstanceMetaDataRepository instanceMetaDataRepository;
 
+    @Value("${cb.security.keypair.cache.size:10}")
+    private int keyPairCacheSize;
+
+    private FixedSizePreloadCache<KeyPair> keyPairCache;
+
+    @PostConstruct
+    public void init() {
+        keyPairCache = new FixedSizePreloadCache<>(keyPairCacheSize, PkiUtil::generateKeypair);
```

Comment:
Instead of postconstructing here, it could be created as a bean in the `FixedSizePreloadCacheConfig` class. where the `@Value("${cb.security.keypair.cache.size:10}")` could be stored as well, it fits there fine.

Reply:
I didn't want to create a bean which is used only one place, here.